xbmc/addons/Addon.h

```diff
@@ -59,7 +59,10 @@ class CAddon : public IAddon
    * @param[in] type The to checked type identifier
    * @return true in case the wanted type is supported, false if not
    */
-  bool HasType(TYPE type) const override { return m_addonInfo->HasType(type); }
+  bool HasType(TYPE type, bool mainOnly) const override
```

Comment:
May I suggest to not add another (unreadable bool) parameter here, but to:

a) Rename `HasType`to `ProvidesType`with semantics of current `HasType(false)`
b) Introduce `GetMainType`with the semantics of current `HasType(true)`

IMO, this makes the calling code far more readable/understandable.

Reply:
yes, the bool params.. i intended to quickly get this fix in and later change the bool param to a `CheckMainOnly::YES/NO`-enum. but i'm also fine with options a) and b)... what would you like more?